src/physfs/util.cpp

```diff
@@ -60,6 +60,21 @@ bool remove(const std::string& filename)
   return PHYSFS_delete(filename.c_str()) == 0;
 }
 
+void
+remove_with_content(std::string& filename)
+{
+  char** files = PHYSFS_enumerateFiles(filename.c_str());
+  for (const char* const* file = files; *file != nullptr; file++)
+  {
+    std::string path = FileSystem::join(filename, *file);
+    if (is_directory(path))
+      remove_with_content(path);
+    PHYSFS_delete(path.c_str());
+  }
+  PHYSFS_freeList(files);
+  PHYSFS_delete(filename.c_str());
+}
```

Comment:
Small detail, but it would be preferable to keep `free` functions at the very end, so that if anyone else comes to maintain the function, they can immediately spot the manually managed lifetime.